From da34ed679d35a425a910301e3b31814b02861130 Mon Sep 17 00:00:00 2001
From: Alan Tull <r80115@freescale.com>
Date: Tue, 11 Nov 2008 14:38:50 -0600
Subject: [PATCH] ENGR00098567 Move mxc_init_audio to mx51_3stack board file

Move imx-3stack-wm8903 and its init code from devices.c to
the imx51_3stack board file as it is board-specific.

Signed-off-by: Alan Tull <r80115@freescale.com>
---
 arch/arm/mach-mx51/devices.c     |   27 ---------------------------
 arch/arm/mach-mx51/mx51_3stack.c |   34 ++++++++++++++++++++++++++++++++++
 2 files changed, 34 insertions(+), 27 deletions(-)

diff --git a/arch/arm/mach-mx51/devices.c b/arch/arm/mach-mx51/devices.c
index 910db17..9f23ae1 100644
--- a/arch/arm/mach-mx51/devices.c
+++ b/arch/arm/mach-mx51/devices.c
@@ -771,32 +771,6 @@ static inline void mxc_init_dma(void)
 	(void)platform_device_register(&mxc_dma_device);
 }
 
-static struct mxc_audio_platform_data mxc_audio_data;
-
-static struct platform_device mxc_alsa_device = {
-	.name = "imx-3stack-wm8903",
-	.id = 0,
-	.dev = {
-		.release = mxc_nop_release,
-		.platform_data = &mxc_audio_data,
-		},
-};
-
-static void mxc_init_audio(void)
-{
-	mxc_audio_data.ssi_clk[0] = clk_get(NULL, "ssi_clk.0");
-	clk_put(mxc_audio_data.ssi_clk[0]);
-
-	mxc_audio_data.ssi_clk[1] = clk_get(NULL, "ssi_clk.1");
-	clk_put(mxc_audio_data.ssi_clk[1]);
-
-	mxc_audio_data.ssi_num = 1;
-	mxc_audio_data.src_port = 2;
-	mxc_audio_data.ext_port = 3;
-
-	platform_device_register(&mxc_alsa_device);
-}
-
 static struct resource spdif_resources[] = {
 	{
 	 .start = SPDIF_BASE_ADDR,
@@ -857,7 +831,6 @@ static int __init mxc_init_devices(void)
 	mxc_init_owire();
 	mxc_init_ipu();
 	mxc_init_vpu();
-	mxc_init_audio();
 	mxc_init_spdif();
 	mxc_init_tve();
 	mx51_init_lpmode();
diff --git a/arch/arm/mach-mx51/mx51_3stack.c b/arch/arm/mach-mx51/mx51_3stack.c
index 93f4308..eee25e5 100644
--- a/arch/arm/mach-mx51/mx51_3stack.c
+++ b/arch/arm/mach-mx51/mx51_3stack.c
@@ -808,6 +808,39 @@ static int __init mxc_init_srpgconfig(void)
 	return 0;
 }
 
+#if defined(CONFIG_SND_SOC_IMX_3STACK_WM8903) \
+    || defined(CONFIG_SND_SOC_IMX_3STACK_WM8903_MODULE)
+static struct mxc_audio_platform_data mxc_audio_data;
+
+static struct platform_device mxc_alsa_device = {
+	.name = "imx-3stack-wm8903",
+	.id = 0,
+	.dev = {
+		.release = mxc_nop_release,
+		.platform_data = &mxc_audio_data,
+		},
+};
+
+static void __init mxc_init_audio(void)
+{
+	mxc_audio_data.ssi_clk[0] = clk_get(NULL, "ssi_clk.0");
+	clk_put(mxc_audio_data.ssi_clk[0]);
+
+	mxc_audio_data.ssi_clk[1] = clk_get(NULL, "ssi_clk.1");
+	clk_put(mxc_audio_data.ssi_clk[1]);
+
+	mxc_audio_data.ssi_num = 1;
+	mxc_audio_data.src_port = 2;
+	mxc_audio_data.ext_port = 3;
+
+	(void)platform_device_register(&mxc_alsa_device);
+}
+#else
+static void __init mxc_init_audio(void)
+{
+}
+#endif
+
 /*!
  * Board specific fixup function. It is called by \b setup_arch() in
  * setup.c file very early on during kernel starts. It allows the user to
@@ -872,6 +905,7 @@ static void __init mxc_board_init(void)
 
 #endif
 	mxc_init_touchscreen();
+	mxc_init_audio()
 
 }
 
-- 
1.5.4.4

